### PR TITLE
Development pagination sorting swagger hw

### PR DESCRIPTION
**Pull request by Alexander1iubchenko**

### CODE REVIEW

File: src/main/java/dto/CreateBookRequestDto.java [hunk 1]
```diff
@@ -1,14 +1,21 @@
 package dto;
 
 import java.math.BigDecimal;
+import javax.validation.constraints.Min;
 import lombok.Data;
+import lombok.NonNull;
 
 @Data
 public class CreateBookRequestDto {
     private Long id;
```
Comment (RostyslavKuzmych) [inline src/main/java/dto/CreateBookRequestDto.java]: Are you sure, that we need this field in the dto?

File: src/main/java/dto/CreateBookRequestDto.java [hunk 2]
```diff
@@ -1,14 +1,21 @@
...
+    @NonNull
     private String title;
+    @NonNull
     private String author;
+    @NonNull
     private String isbn;
+    @NonNull
+    @Min(0)
```
Comment (RostyslavKuzmych) [inline src/main/java/dto/CreateBookRequestDto.java]: You use annotations of lombok, and don't use annotations from package of validation.

File: src/main/java/controller/BookController.java [hunk 3]
```diff
@@ -25,28 +29,33 @@ public class BookController {
...
     @DeleteMapping("/{id}")
+    @Operation(summary = "Delete a book by id", description = "Delete a book by id in database")
     public void deleteBook(@PathVariable Long id) {
         bookService.deleteById(id);
     }
 
     @ResponseStatus(HttpStatus.ACCEPTED)
     @PutMapping("/{id}")
+    @Operation(summary = "Update a book by id", description = "Update a book by id in database")
     public void updateBook(@PathVariable Long id, @RequestBody CreateBookRequestDto requestDto) {
```
Comment (RostyslavKuzmych) [inline src/main/java/controller/BookController.java]: Where is annotation Valid?

File: src/main/java/controller/BookController.java [hunk 4]
```diff
@@ -25,28 +29,33 @@ public class BookController {
...
     }
 
     @GetMapping("/{id}")
+    @Operation(summary = "Get a book by id", description = "Gets a book by id")
     public BookDto getBookById(@PathVariable Long id) {
         return bookService.getBookById(id);
     }
 
     @PostMapping
+    @Operation(summary = "Create a new book", description = "Creates a new book")
```
Comment (RostyslavKuzmych) [inline src/main/java/controller/BookController.java:45]: the same

File: src/main/java/dto/CreateBookRequestDto.java [hunk 5]
```diff
@@ -1,14 +1,21 @@
...
 @Data
 public class CreateBookRequestDto {
     private Long id;
+    @NonNull
     private String title;
+    @NonNull
     private String author;
+    @NonNull
     private String isbn;
+    @NonNull
```
Comment (RostyslavKuzmych) [inline src/main/java/dto/CreateBookRequestDto.java]: Don't use annotation NonNull and Min simultaneously

**Review by RostyslavKuzmych — CHANGES_REQUESTED** [review]
Overall, good. But you must analyze my comments.

File: src/main/java/dto/CreateBookRequestDto.java [hunk 6]
```diff
@@ -1,14 +1,21 @@
...
 
 import java.math.BigDecimal;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
 import lombok.Data;
 
 @Data
 public class CreateBookRequestDto {
     private Long id;
+    @NotNull
```
Comment (bhdnchui) [inline src/main/java/dto/CreateBookRequestDto.java]: Define messages for validation annotations.
For example: `@NotNull(message = "Title cannot be null.")`

File: src/main/java/dto/CreateBookRequestDto.java (hunk 1 above)
Comment (bhdnchui) [reply src/main/java/dto/CreateBookRequestDto.java]: Yeap, you can not request to create a book and define id of it. 
The server-side should create a book and define id for that book.

**Review by bhdnchui — CHANGES_REQUESTED** [review]
Just a few points to consider.

File: pom.xml [hunk 7]
```diff
@@ -68,6 +73,12 @@
             <artifactId>mapstruct</artifactId>
             <version>1.5.5.Final</version>
         </dependency>
+
+        <dependency>
+            <groupId>jakarta.validation</groupId>
```
Comment (vovasalyha) [inline pom.xml:78]: If I'm not mistaken, this dependency is fetched by the validation starter.

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**